pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/api/ClassMetricKey.java

```diff
@@ -45,8 +65,41 @@ public ClassMetric getCalculator() {
     }
 
 
+    @Override
     public boolean supports(ASTAnyTypeDeclaration node) {
         return calculator.supports(node);
     }
 
+
+    /**
+     * Creates a new metric key holding a metric which can be computed on a class.
+     *
+     * TODO:cf Generify and move to MetricKey after upgrading compiler to 1.8
+     *
+     * @param metric The metric to use
+     * @param name   The name of the metric
+     *
+     * @return The metric key
+     */
+    public static MetricKey<ASTAnyTypeDeclaration> of(final Metric<ASTAnyTypeDeclaration> metric, final String name) {
```

Comment:
I think it could be part of the public API of the framework, to ease the use of custom metrics